### DISCUSSION TITLE
WinUI template: Accessibility for existing elements like SoftwareBitmap etc.

### DISCUSSION THREAD

**Discussion by jp-weber** [unoplatform/uno#11189] [General]
Access for MediaPlayerElement and related elements for the WinUI template (4.8 preview) is being enabled. I wonder if this is also planned for the following elements?

- SoftwareBitmap
- BitmapEncoder
- MediaSource
- etc.

**Comment by jeromelaban** [comment]
Thanks for the question. Can you provide more information about what you're looking for? Which scenario is not working?

**Reply by jp-weber** [reply]
Thank you for the fast answer. I am developing a two-factor authenticator and would like to continue using the existing code as much as possible for my uno port from the UWP base:
https://github.com/2fast-team/2fast

The SoftwareBitmap is used in my app mainly to read the QR code using ZXing.Net:
https://github.com/2fast-team/2fast/blob/master/Project2FA/Project2FA.UWP/ViewModels/ContentDialogs/AddAccountContentDialogViewModel.cs#L565
With the WinUI template the SoftwareBitmap is not accessible, with the older Uno.UI, however, it does. I know that currently the BitmapDecoder and other used elements are not implemented in Uno, but it is gradually becoming the perfect platform if you are coming from UWP and want to develop for other platforms :)

**Reply by jeromelaban** [reply]
Thanks for the details! `SoftwareBitmap` is implemented for [some platforms](https://github.com/unoplatform/uno/blob/510d75bd0ead90399208905f28d7c6dbfa13ab23/src/Uno.UWP/Graphics/Imaging/SoftwareBitmap.skia.cs#L55). What error are you getting when trying to use `SoftwareBitmap` ?

You can also use `WriteableBitmap`, depending on the features that you are working with.

**Reply by jp-weber** [reply]
I will get an error with not implemented, if I go to the definition I see this (net7.0-ios):
![grafik](https://user-images.githubusercontent.com/10946850/217658263-d953d9a5-c9bd-449e-84b0-298d32d15731.png)

WriteableBitmap would of course be an alternative if this were changed throughout.

**Reply by jeromelaban** [reply]
I see. This is only a subset of properties that are not available. Have you tried using the rest of the class (assuming that the DPI is 96.0)?
